### ISSUE TITLE
AI Rogue backstabs its own minion

### ISSUE THREAD

**Issue by jleclanche** [HearthSim/hs-bugs#136]
Board state:

* Player
  * Grim Patron, 2 health
  * Grim Patron, 2 health
  * Acolyte of Pain, 3 health
* Innkeeper (Normal deck, rogue)
  * Empty board

AI plays the following:
  * Nightblade
  * Backstab on OWN Nightblade
  * Sap on Acolyte of Pain

**Comment by jleclanche** [comment]
Full log of the turn:

```
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_NEXT
ACTION_START - Entity=1 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=2 tag=NUM_TURNS_LEFT value=0
        TAG_CHANGE - Entity=2 tag=NUM_TURNS_LEFT value=1
        TAG_CHANGE - Entity=3 tag=CURRENT_PLAYER value=0
        TAG_CHANGE - Entity=2 tag=CURRENT_PLAYER value=1
        TAG_CHANGE - Entity=GameEntity tag=TURN value=15
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_READY
ACTION_END
TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=ATK value=1
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_READY
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=NUM_TURNS_IN_PLAY value=8
        TAG_CHANGE - Entity=[name= id=37 cardId=CS2_083b] tag=NUM_TURNS_IN_PLAY value=8
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=NUM_TURNS_IN_PLAY value=1
        TAG_CHANGE - Entity=2 tag=RESOURCES value=8
        TAG_CHANGE - Entity=2 tag=RESOURCES_USED value=0
        TAG_CHANGE - Entity=2 tag=COMBO_ACTIVE value=0
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_PLAYED_THIS_TURN value=0
        TAG_CHANGE - Entity=2 tag=NUM_MINIONS_PLAYED_THIS_TURN value=0
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=0
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=EXHAUSTED value=0
        TAG_CHANGE - Entity=[name= id=37 cardId=CS2_083b] tag=EXHAUSTED value=0
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=EXHAUSTED value=0
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_DRAWN_THIS_TURN value=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_START_TRIGGERS
        TAG_CHANGE - Entity=2 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=0
        TAG_CHANGE - Entity=3 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=0
ACTION_END
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_START_TRIGGERS
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_START
ACTION_END
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_START
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=66 tag=ZONE value=HAND
        TAG_CHANGE - Entity=66 tag=ZONE_POSITION value=6
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_DRAWN_THIS_TURN value=1
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=NUM_ATTACKS_THIS_TURN value=0
        TAG_CHANGE - Entity=2 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=0
        TAG_CHANGE - Entity=3 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=0
        TAG_CHANGE - Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=0
        TAG_CHANGE - Entity=2 tag=HEROPOWER_ACTIVATIONS_THIS_TURN value=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_ACTION
ACTION_END
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_ACTION
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_END
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=9
ACTION_START - Entity=40 BlockType=PLAY Index=0 Target=0
        TAG_CHANGE - Entity=2 tag=RESOURCES_USED value=5
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_PLAYED_THIS_TURN value=1
        TAG_CHANGE - Entity=2 tag=NUM_MINIONS_PLAYED_THIS_TURN value=1
        TAG_CHANGE - Entity=66 tag=ZONE_POSITION value=5
        TAG_CHANGE - Entity=47 tag=ZONE_POSITION value=4
        TAG_CHANGE - Entity=53 tag=ZONE_POSITION value=3
        TAG_CHANGE - Entity=44 tag=ZONE_POSITION value=2
        TAG_CHANGE - Entity=59 tag=ZONE_POSITION value=1
        SHOW_ENTITY - Updating Entity=40 CardID=EX1_593
                tag=HEALTH value=4
                tag=ATK value=4
                tag=COST value=5
                tag=ZONE value=PLAY
                tag=FACTION value=NEUTRAL
                tag=CARDTYPE value=MINION
                tag=RARITY value=FREE
                tag=BATTLECRY value=1
        TAG_CHANGE - Entity=40 tag=EXHAUSTED value=1
        TAG_CHANGE - Entity=40 tag=JUST_PLAYED value=1
        TAG_CHANGE - Entity=2 tag=LAST_CARD_PLAYED value=40
        ACTION_START - Entity=40 BlockType=POWER Index=-1 Target=0
                META_DATA - Meta=TARGET Data=0 Info=1
                        Info[0] = 4
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=PREDAMAGE value=3
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=PREDAMAGE value=0
                META_DATA - Meta=DAMAGE Data=3 Info=1
                        Info[0] = 4
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=LAST_AFFECTED_BY value=40
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=DAMAGE value=7
        ACTION_END
        TAG_CHANGE - Entity=2 tag=COMBO_ACTIVE value=1
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=1
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=6
ACTION_START - Entity=66 BlockType=PLAY Index=0 Target=40
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_PLAYED_THIS_TURN value=2
        SHOW_ENTITY - Updating Entity=66 CardID=CS2_072
                tag=ZONE value=PLAY
                tag=FACTION value=NEUTRAL
                tag=CARDTYPE value=ABILITY
                tag=RARITY value=FREE
        TAG_CHANGE - Entity=66 tag=ZONE_POSITION value=0
        TAG_CHANGE - Entity=66 tag=JUST_PLAYED value=1
        TAG_CHANGE - Entity=2 tag=LAST_CARD_PLAYED value=66
        TAG_CHANGE - Entity=66 tag=CARD_TARGET value=40
        ACTION_START - Entity=66 BlockType=POWER Index=-1 Target=40
                META_DATA - Meta=TARGET Data=0 Info=1
                        Info[0] = 40
                TAG_CHANGE - Entity=40 tag=PREDAMAGE value=2
                TAG_CHANGE - Entity=40 tag=PREDAMAGE value=0
                META_DATA - Meta=DAMAGE Data=2 Info=1
                        Info[0] = 40
                TAG_CHANGE - Entity=40 tag=LAST_AFFECTED_BY value=66
                TAG_CHANGE - Entity=40 tag=DAMAGE value=2
        ACTION_END
        TAG_CHANGE - Entity=66 tag=ZONE value=GRAVEYARD
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=2
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=5
ACTION_START - Entity=59 BlockType=PLAY Index=0 Target=30
        TAG_CHANGE - Entity=2 tag=RESOURCES_USED value=7
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_PLAYED_THIS_TURN value=3
        TAG_CHANGE - Entity=47 tag=ZONE_POSITION value=3
        TAG_CHANGE - Entity=53 tag=ZONE_POSITION value=2
        TAG_CHANGE - Entity=44 tag=ZONE_POSITION value=1
        SHOW_ENTITY - Updating Entity=59 CardID=EX1_581
                tag=COST value=2
                tag=ZONE value=PLAY
                tag=FACTION value=NEUTRAL
                tag=CARDTYPE value=ABILITY
                tag=RARITY value=FREE
        TAG_CHANGE - Entity=59 tag=ZONE_POSITION value=0
        TAG_CHANGE - Entity=59 tag=JUST_PLAYED value=1
        TAG_CHANGE - Entity=2 tag=LAST_CARD_PLAYED value=59
        TAG_CHANGE - Entity=59 tag=CARD_TARGET value=30
        ACTION_START - Entity=59 BlockType=POWER Index=-1 Target=30
                META_DATA - Meta=TARGET Data=0 Info=1
                        Info[0] = 30
                TAG_CHANGE - Entity=30 tag=ZONE value=HAND
                TAG_CHANGE - Entity=30 tag=ZONE_POSITION value=9
        ACTION_END
        TAG_CHANGE - Entity=59 tag=ZONE value=GRAVEYARD
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=3
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=3
ACTION_START - Entity=36 BlockType=ATTACK Index=-1 Target=78
        TAG_CHANGE - Entity=GameEntity tag=PROPOSED_ATTACKER value=36
        TAG_CHANGE - Entity=GameEntity tag=PROPOSED_DEFENDER value=78
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=ATTACKING value=1
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_ACTION
        TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_COMBAT
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=4
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=DEFENDING value=1
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=PREDAMAGE value=1
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=PREDAMAGE value=0
        META_DATA - Meta=DAMAGE Data=1 Info=1
                Info[0] = 78
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=LAST_AFFECTED_BY value=36
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=DAMAGE value=3
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=PREDAMAGE value=3
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=PREDAMAGE value=0
        META_DATA - Meta=DAMAGE Data=3 Info=1
                Info[0] = 36
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=LAST_AFFECTED_BY value=78
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=DAMAGE value=26
        ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
                TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=PREDAMAGE value=1
                TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=PREDAMAGE value=0
                META_DATA - Meta=DAMAGE Data=1 Info=1
                        Info[0] = 98
                TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=DAMAGE value=2
        ACTION_END
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=NUM_ATTACKS_THIS_TURN value=1
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=EXHAUSTED value=1
        TAG_CHANGE - Entity=GameEntity tag=PROPOSED_ATTACKER value=0
        TAG_CHANGE - Entity=GameEntity tag=PROPOSED_DEFENDER value=0
        TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=ATTACKING value=0
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=DEFENDING value=0
ACTION_END
ACTION_START - Entity=1 BlockType=DEATHS Index=0 Target=0
        TAG_CHANGE - Entity=3 tag=NUM_MINIONS_PLAYER_KILLED_THIS_TURN value=1
        TAG_CHANGE - Entity=GameEntity tag=NUM_MINIONS_KILLED_THIS_TURN value=1
        TAG_CHANGE - Entity=[name= id=84 cardId=BRM_019] tag=ZONE_POSITION value=2
        TAG_CHANGE - Entity=[name= id=81 cardId=BRM_019] tag=ZONE_POSITION value=1
        TAG_CHANGE - Entity=3 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_TURN value=1
        TAG_CHANGE - Entity=3 tag=NUM_FRIENDLY_MINIONS_THAT_DIED_THIS_GAME value=5
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=ZONE value=GRAVEYARD
        TAG_CHANGE - Entity=[name= id=79 cardId=EX1_084e] tag=ATTACHED value=0
        TAG_CHANGE - Entity=[name= id=79 cardId=EX1_084e] tag=ZONE value=REMOVEDFROMGAME
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=IGNORE_DAMAGE value=1
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=DAMAGE value=0
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=IGNORE_DAMAGE_OFF value=1
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=CHARGE value=0
        TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=ZONE_POSITION value=0
        TAG_CHANGE - Entity=2 tag=334 value=0
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=ZONE value=GRAVEYARD
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=IGNORE_DAMAGE value=1
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=DAMAGE value=0
        TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=IGNORE_DAMAGE_OFF value=1
ACTION_END
TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=AI_ONE_SHOT_KILL value=0
TAG_CHANGE - Entity=[name= id=36 cardId=HERO_03] tag=ATK value=0
TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=IGNORE_DAMAGE value=0
TAG_CHANGE - Entity=[name= id=78 cardId=BRM_019] tag=IGNORE_DAMAGE_OFF value=0
TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=IGNORE_DAMAGE value=0
TAG_CHANGE - Entity=[name= id=98 cardId=CS2_082] tag=IGNORE_DAMAGE_OFF value=0
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_ACTION
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_END
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=2
ACTION_START - Entity=53 BlockType=PLAY Index=0 Target=0
        TAG_CHANGE - Entity=2 tag=RESOURCES_USED value=8
        TAG_CHANGE - Entity=2 tag=NUM_CARDS_PLAYED_THIS_TURN value=4
        TAG_CHANGE - Entity=47 tag=ZONE_POSITION value=2
        SHOW_ENTITY - Updating Entity=53 CardID=CS2_075
                tag=COST value=1
                tag=ZONE value=PLAY
                tag=FACTION value=NEUTRAL
                tag=CARDTYPE value=ABILITY
                tag=RARITY value=FREE
        TAG_CHANGE - Entity=53 tag=ZONE_POSITION value=0
        TAG_CHANGE - Entity=53 tag=JUST_PLAYED value=1
        TAG_CHANGE - Entity=2 tag=LAST_CARD_PLAYED value=53
        ACTION_START - Entity=53 BlockType=POWER Index=-1 Target=0
                META_DATA - Meta=TARGET Data=0 Info=1
                        Info[0] = 4
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=PREDAMAGE value=3
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=PREDAMAGE value=0
                META_DATA - Meta=DAMAGE Data=3 Info=1
                        Info[0] = 4
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=LAST_AFFECTED_BY value=53
                TAG_CHANGE - Entity=[name= id=4 cardId=HERO_01a] tag=DAMAGE value=10
        ACTION_END
        TAG_CHANGE - Entity=53 tag=ZONE value=GRAVEYARD
        TAG_CHANGE - Entity=2 tag=NUM_OPTIONS_PLAYED_THIS_TURN value=5
ACTION_END
TAG_CHANGE - Entity=2 tag=NUM_OPTIONS value=1
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_END
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_CLEANUP
ACTION_END
TAG_CHANGE - Entity=GameEntity tag=STEP value=MAIN_CLEANUP
ACTION_START - Entity=2 BlockType=TRIGGER Index=-1 Target=0
        TAG_CHANGE - Entity=40 tag=JUST_PLAYED value=0
        TAG_CHANGE - Entity=GameEntity tag=NEXT_STEP value=MAIN_NEXT
ACTION_END
```

**Comment by Patashu** [comment]
I bet the logic was that it wanted to Backstab something, but then realized the only enemy target it didn't want to Backstab (as you'd draw a card), but it didn't then abort the Backstabbing and did it on the Nightblade instead.

**Comment by culinko** [comment]
There are literally dozens of similar bugs like this. The AI just plays the card because the condition to play it is met (like Druid AI brewing the brew again and again until he hits 10 mana).

**Comment by culinko** [comment]
Merged to #8.
